docs/docs/core_concepts.md

```diff
@@ -1,94 +1,115 @@
 
-On the frontend side of things TwigYard is pretty technology agnostic. Type raw css, use grunt, gulp or whatever you like. You do not need to modify your devstack in any way. Within the site folder you are free to use any tools you like to as long as they generate the required files.
+On the frontend side of things TwigYard is pretty technology agnostic. Type raw css, use grunt, gulp or whatever you like. You do not need to modify your current devstack in any way to use TwigYard. Within the site folder you are free to use any tools you like to as long as they generate the required file structure.
 
 ## site.yml
 TwigYard requires that each site has a main config file whose name is defined in the instance wide `parameters.yml` file. While the name can be anything, it is recommended to name it `site.yml` if possible. For simplicity we will refer to this main site config as `site.yml` in this manual.
 
-See the middleware reference for supported configuration options.
+The `site.yml` has two versions. The now deprecated version 1 functions as the fallback if no version is specified. See the middleware reference for supported configuration options.
 
-### Includes
-The `site.yml` can include other files to make the configuration more flexible.
-This feature has two benefits.
-
-First, it allows the config to be broken up into multiple files for better readability.
+### Substitution
+Some data needed in the `site.yml` file might be to sensitive to be commited to VCS. These secrets can be defined in a file `parameters.yml` which can be kept out of version control. Data from this file can then be included in the`site.yml` via the `%param_name%` syntax. All the parameters to be substituted must be defined in the `parameters.yml` as a flat map under top level `parameters` key. The file must be manually included in the `site.yml`. 
+ 
+### Composition
+The `site.yml` can include other files to make the configuration more flexible. For example it allows the config to be broken up into multiple files for better readability.
 ```
+# site.yml
+
 ...
 imports:
     - { resource: 'router.yml' }
     - { resource: 'renderer.yml' }
 ```
-Second, it makes it possible to have the site behave differently on different instances. If, for example, we would like to have a staging instance where the site must be protected by httpauth header, we would have the following in `site_staging.yml` and set the instance to use this is the main config file.
+
+### Overloading
+Imports also make it possible to have the site behave differently on different environments. On a staging instance, for example, we might prefer to use different configuration. Typically we need to protect the site by httpauth. This can be solved by creating a separate `site_staging.yml` which imports the production `site.yml` and set the staging instance to use this is the main config file.
```

Comment:
Please rewrite whole sentence, it does not make sense to much. `Imports also make it possible to have the site behave differently on different environments`